[PyTorchEdge] backport v8 to v7 to support promoted ops as instruction

pavithranrao opened the pull request:
Backport v8 to v7 to support promoted ops as instruction

**Summary**:
_Version bump to V8_.

**Changes introduced**:

1. Promoted ops (ops from lookup table) are made as instructions
2. RaiseException now supports an additional optional argument that provides the exception message to debug
3. V7 and below will continue to use operators and V8 an above will use instructions.
4. Backport logic to help with BC




Differential Revision: D33719098

iseeyuan commented on torch/csrc/jit/runtime/interpreter/code_impl.h:
```diff
@@ -890,10 +904,25 @@ struct MobileCodeImpl : CodeImpl {
     }
   }
 
+  void emitOperatorOrInstruction(
```
@zhxchen17 , could help review the emitting part?

cccclai commented:
LGTM. Thanks! Could you also update the module https://github.com/pytorch/pytorch/blob/deb1c2f8370714ab32f5416d1fd86daa07906a88/test/cpp/jit/test_lite_interpreter.cpp#L629-L645 and make it also covers on of the promoted ops?

iseeyuan reviewed (approved):
LGTM. Please make sure test all production models
1. In v8
2. backport to v7

pavithranrao committed:
[PyTorchEdge] backport v8 to v7 to support promoted ops as instruction (#71662)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/71662

backport v8 to v7 to support promoted ops as instruction

a flag to help export as instruction from v8 and export as operators for v7 and below

Test Plan:
```
buck test caffe2/test/cpp/jit:jit -- LiteInterpreterTest.BackPortByteCodeModelAllVersions

Started reporting to test run: https://www.internalfb.com/intern/testinfra/testrun/5629499620570927
    ✓ ListingSuccess: caffe2/test/cpp/jit:jit : 461 tests discovered (15.693)
    ✓ Pass: caffe2/test/cpp/jit:jit - LiteInterpreterTest.BackPortByteCodeModelAllVersions (2.712)
Summary
  Pass: 1
  ListingSuccess: 1
If you need help understanding your runs, please follow the wiki: https://fburl.com/posting_in_tpx_users
Finished test run: https://www.internalfb.com/intern/testinfra/testrun/5629499620570927
```

```
buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen

buck test mode/opt //caffe2/test:upgrader_codegen -- mobile.test_upgrader_codegen.TestLiteScriptModule
Parsing buck files: finished in 0.8 sec
Downloaded 0/2 artifacts, 0.00 bytes, 100.0% cache miss (for updated rules)
Building: finished in 01:39.4 min (100%) 11031/11031 jobs, 2/11031 updated
  Total time: 01:40.2 min
More details at https://www.internalfb.com/intern/buck/build/a8b0e417-019c-44ba-be6b-23379411a965
BUILD SUCCEEDED
Tpx test run coordinator for Facebook. See https://fburl.com/tpx for details.
Running with tpx session id: 44fbfa66-cce8-4277-82ac-f89d79558581
Trace available for this run at /tmp/tpx-20220202-160956.915412/trace.log
RemoteExecution session id: reSessionID-44fbfa66-cce8-4277-82ac-f89d79558581-tpx
Started reporting to test run: https://www.internalfb.com/intern/testinfra/testrun/281475200877601
    ✓ ListingSuccess: caffe2/test:upgrader_codegen : 1 tests discovered (1.249)
    ✓ Pass: caffe2/test:upgrader_codegen - test_generate_bytecode (mobile.test_upgrader_codegen.TestLiteScriptModule) (1.365)
Summary
  Pass: 1
  ListingSuccess: 1
If you need help understanding your runs, please follow the wiki: https://fburl.com/posting_in_tpx_users
Finished test run: https://www.internalfb.com/intern/testinfra/testrun/281475200877601
```

Reviewed By: iseeyuan

Differential Revision: D33719098

fbshipit-source-id: 93a25e69e7d3c29f027e4f9d04fdb1c27cbe2400
